src/io/sb3/fromSb3.ts

```diff
@@ -416,11 +360,12 @@ export async function fromSb3JSON(json: sb3.ProjectJSON, options: { getAsset: ge
   return new Project({
     stage: new Stage({
       name: stage.name,
-      costumes: await extractCostumes(stage, options.getAsset),
+      costumes: await extractCostumes(stage, options.getAssetData),
       costumeNumber: stage.currentCostume,
-      sounds: await extractSounds(stage, options.getAsset),
+      sounds: await extractSounds(stage, options.getAssetData),
       scripts: Object.entries(stage.blocks)
         .filter(([id, block]) => block.topLevel)
+        .filter(([id, block]) => !block.shadow)
```

Comment:
```diff
-        .filter(([id, block]) => block.topLevel)
-        .filter(([id, block]) => !block.shadow)
+        .filter(([id, block]) => block.topLevel && !block.shadow)
```
Half the filtering, twice\* the performance!


\* 60% of the time, it increases performance every time